pkg/testgridanalysis/testgridconversion/to_raw_data_test.go

```diff
@@ -0,0 +1,574 @@
+		assertNoWarnings(t, opts.warnings)
+	}
+
+	for _, testName := range unignoredTestNames {
+		// Check that unignored test names are, indeed, not ignored.
+		t.Run("UnignoredRegex"+testName, func(t *testing.T) {
+			if testgridconversion.IsIgnoredTest(testName) {
+				t.Errorf("expected %s not to be ignored", testName)
+			}
+		})
+
+		// Run the raw data test case to verify that non-ignored tests are found
+		t.Run("Unignored", func(t *testing.T) {
+			runRawDataTestCase(t, rawDataTestCase{
+				testGridTestNames: []string{testName},
+				expectedTestNames: []string{testName},
+				options:           options,
+				testFunc:          testFunc,
+			})
+		})
+	}
+
+	for _, testName := range ignoredTestNames {
+		// Create a sub-test per ignored test name
+		t.Run(testName, func(t *testing.T) {
+			// Check that the regex matches the ignored test name
+			t.Run("Regex", func(t *testing.T) {
+				if !testgridconversion.IsIgnoredTest(testName) {
+					t.Errorf("expected %s to be ignored", testName)
+				}
+			})
+
+			// Run the raw data test case to verify that no unexpected tests are found
+			runRawDataTestCase(t, rawDataTestCase{
+				testGridTestNames: []string{testName},
+				expectedTestNames: []string{},
+				options:           options,
+				testFunc:          testFunc,
+			})
+		})
+	}
+}
+
+func TestToRawDataSynthetics(t *testing.T) {
+	// These test cases are intended to test the interaction of ToRawData with
+	// the Openshift Synthetic Test Manager.
+
+	// These will be appended to the expected test names for each test case below
+	sippySyntheticTests := []string{
+		"[sig-sippy] infrastructure should work",
+		"[sig-sippy] install should not timeout",
+		"[sig-sippy] install should work",
+		"[sig-sippy] openshift-tests should work",
+		"[sig-sippy] tests should finish with healthy operators",
+	}
+
+	missingSetupJobWarning := []string{
+		fmt.Sprintf("\"%s\" is missing a test setup job to indicate successful installs", jobName),
+	}
+
+	testCases := []struct {
+		name              string
+		testGridTestNames []string
+		expectedTestNames []string
+		// These are grouped by test status because certain test statuses (success,
+		// failure, flake) will cause the OpenshiftSyntheticTestManager emit a
+		// warning under specific cases.
+		expectedWarnings map[testgridv1.TestStatus][]string
+	}{
+		// A job with no synthetic tests should still have synthetic test outcomes
+		// when the synthetic test manager is used.
+		{
+			name: "nonsynthetic test",
+			testGridTestNames: []string{
+				"not a synthetic test",
+			},
+			expectedTestNames: []string{
+				"not a synthetic test",
+			},
+			// We expect these warnings to be present because the "container setup"
+			// or "Overall" tests were not run.
+			expectedWarnings: map[testgridv1.TestStatus][]string{
+				testgridv1.TestStatusSuccess: missingSetupJobWarning,
+				testgridv1.TestStatusFailure: missingSetupJobWarning,
+				testgridv1.TestStatusFlake:   missingSetupJobWarning,
```

Comment:
i'm confused about why this is expected to be emitted as successful, failing, and flaking, simultaneously for a given run?

Reply:
It's not expecting that from a single run of `ProcessTestGridDataIntoRawJobResults`. The testcase calls `ProcessTestGridDataIntoRawJobResults` three times, once with each of those TestGrid statuses being set. This groups the expected warnings by the TestGrid status.

Reply:
why doesn't the `Overall` testcase a little further down also have all 3 of these then?

on `Overall` sucess/flake we don't warn about missing setup, but if there's no `Overall` test we always warn about the missing setup?

Reply:
I can explain why the test passes as-is, but I cannot say if the behavior the test verifies is what is desired. The why behind it is:

In the case where `Overall` is a success, `processTestToJobRunResults` flips the success bit and sets `SetupStatus`: https://github.com/openshift/sippy/blob/master/pkg/testgridanalysis/testgridconversion/to_raw_data.go#L144-L148.

In the case where `Overall` fails, `processTestToJobRunResults` flips the failure bit, but it does not set `SetupStatus`: https://github.com/openshift/sippy/blob/9f8f55fd255647d7f9aacf698e78654e3279ea6c/pkg/testgridanalysis/testgridconversion/to_raw_data.go#L189-L190.

When this code path is hit, the test name "Overall" falls through the subsequent regex checks (https://github.com/openshift/sippy/blob/master/pkg/testgridanalysis/testgridconversion/to_raw_data.go#L196-L206) since it doesn't match any of those cases. If there was another test name that matches one (or more) of those regex tests, `SetupStatus` would get set. For this particular test case, there are no other test names that would pass one of those regex checks, so `SetupStatus` never gets set. Adding another test name that would pass one of those regex checks causes `SetupStatus` to be set, and no warnings are emitted.

Maybe this test is a edge-case that would never actually occur? Should `Overall` also set `SetupStatus`?

Reply:
ok, i think that behavior makes sense.

We can always assume there will be an Overall test (it's possible sippy should throw some other warning/error if it can't find an Overall test result in a run).  And when overall is successful, setting setupstatus to successful makes sense, as described in the comment in the code.

so your test case where there is not an `overall` test result is testing for behavior/situations that we should never encounter in practice.  Again it might be better to have sippy completely error out in such cases, because i'm not sure if there are other aspects of the reporting that will become confusing/meaningless if there was no Overall result in the run.

Reply:
That makes sense. Since we process multiple jobs and their tests within this codepath, we should run a pre-check to ensure that each job has an "Overall" test before we process the individual tests. This way, we can skip jobs that don't have an "Overall" test and still process the ones that do without affecting our statistics. We'll emit one error for each one we find without an "Overall" test since it's possible we could have multiple jobs.

I started coding this up and thankfully, the newly-merged sippyserver smoke tests caught a few edge-cases I need to look into.

Reply:
I've got an initial implementation of this pushed up for review. I was trying to decide what to do with the errors beyond just logging them to the console, so within the UI portion of the server, I emit them using the same path as the synthetic test and bug cache warnings.